File: job-fronted/src/app/services/blog.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { UtilitiesService } from './utilities.service';

@Injectable({
  providedIn: 'root',
})
export class BlogService {
  private url: string;
  constructor(private http: HttpClient, private utilities: UtilitiesService) {
    this.url = utilities.getURLDev();
  }
  blogList(pageIndex: number, pageSize: number): Observable<any> {
    return this.http.get<any>(
      `${this.url}/blog/pub/blogList/${pageIndex}/${pageSize}`
    );
  }

  commentList(pageIndex: number, pageSize: number): Observable<any> {
    return this.http.get<any>(
      `${this.url}/blog/pub/commentList/${pageIndex}/${pageSize}`
    );
  }

  postBlog(blog: any) {
    return this.http.post<any>(
      `${this.url}/blog/pri/postBlog`,
      blog,
      { withCredentials: true }
    );
  }
  blogDetail(blogId: bigint) {
    return this.http.get<any>(
      `${this.url}/blog/pub/blogDetail/${blogId}`
    );
  }
  like(blogId: bigint) {
    return this.http.post<any>(
      `${this.url}/blog/pri/like`,
      blogId,
      { withCredentials: true }
    );
  }
  unlike(blogId: bigint) {
    return this.http.post<any>(
      `${this.url}/blog/pri/unlike`,
      blogId,
      { withCredentials: true }
    );
  }
  comment(comment: any) {
    return this.http.post<any>(
      `${this.url}/blog/pri/comment`,
      comment,
      { withCredentials: true }
    );
  }
}

File: job-backend/src/main/java/com/job_web/service/blog/BlogService.java
package com.job_web.service.blog;

import com.job_web.dto.common.ApiResponse;
import com.job_web.dto.blog.BlogDTO;
import com.job_web.models.Blog;
import com.job_web.models.Comment;
import com.job_web.models.User;
import org.springframework.data.domain.Page;

public interface BlogService {
    String postBlog(BlogDTO blog, User user);
    String updateBlog(long id, BlogDTO blog, User user);
    String deleteBlog(long id);
    Page<Blog> getBlogs(int pageIndex, int pageSize);
    Blog getBlogById(long id);
    String comment(Comment comment, User user);
    String like(long id, User user);
    String unlike(long id, User user);
    Page<Comment> getComments(int pageIndex, int pageSize);
}
